### PR TITLE
Support RuboCop's inherit_from

### PR DESCRIPTION
**Pull request by jakecraige**
https://trello.com/c/hSQ6Xi7l/514-support-rubocop-s-inherit-from

### CODE REVIEW

File: app/models/repo_config.rb [hunk 1]
```diff
@@ -75,6 +75,17 @@ def config_path_for(language)
       hound_config[language]["config_file"]
   end
 
+  def load_config(config_file_path, file_type)
+    main_config = load_file(config_file_path, file_type)
+    inherit_from = main_config.fetch("inherit_from", [])
+
+    inherited_config = inherit_from.reduce({}) do |config, ancestor_file_path|
```
Comment (jakecraige) [inline app/models/repo_config.rb]: Style guide says the opposite "Prefer reduce over inject." [here](https://github.com/thoughtbot/guides/blob/master/style/ruby/README.md#ruby)

Comment (gylaz) [inline app/models/repo_config.rb]: Looks to be a bug in Hound. We should probably remove: https://github.com/thoughtbot/hound/blob/master/.ruby-style.yml#L93

File: app/models/repo_config.rb [hunk 2]
```diff
@@ -75,6 +75,17 @@ def config_path_for(language)
...
+      config.merge(load_file(ancestor_file_path, file_type))
```
Comment (gylaz) [inline app/models/repo_config.rb]: What does this line return? Is it a path?

**Comment by gylaz** [comment]
Nice work! :+1:

File: app/models/repo_config.rb (hunk 2 above)
Comment (jakecraige) [inline app/models/repo_config.rb]: Seems like it's confusing. I've moved to load_file into a variable so clear up what it is in https://github.com/thoughtbot/hound/commit/87ac1382fa69683bad12938ab67d141116424c19

File: app/models/repo_config.rb (hunk 1 above)
Comment (jakecraige) [inline app/models/repo_config.rb]: PR submitted https://github.com/thoughtbot/hound/pull/718

**Comment by salbertson** [comment]
Awesome! @jakecraige, we've been wanting to do this for a while. We should get this merged and test it on staging. :+1:

**Pull request merged**